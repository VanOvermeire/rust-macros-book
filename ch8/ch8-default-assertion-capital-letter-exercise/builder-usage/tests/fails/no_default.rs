use builder_macro::Builder;

struct DoesNotImplementDefault;

#[derive(Builder)]
#[builder_defaults]
struct ExampleStruct {
    not: DoesNotImplementDefault
}

fn main() {}
