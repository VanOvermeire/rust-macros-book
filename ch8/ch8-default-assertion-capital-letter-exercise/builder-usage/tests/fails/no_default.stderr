error[E0277]: the trait bound `DoesNotImplementDefault: Default` is not satisfied
 --> tests/fails/no_default.rs:8:10
  |
8 |     not: DoesNotImplementDefault
  |          ^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `DoesNotImplementDefault`
  |
  = help: see issue #48214
help: consider annotating `DoesNotImplementDefault` with `#[derive(Default)]`
  |
3 + #[derive(Default)]
4 | struct DoesNotImplementDefault;
  |
